Corporate Finance Exercise set 3 Question 4-48: 
to get all the possible IRRs through solving the polynomial equation
3.25*x*(1+x)^4-(1+x)^4+1 = 0, we run the following 3 lines of codes.

In [1]:
from sympy import Symbol, solve
x = Symbol("x")
solve(3.25*x*(1+x)**4-(1+x)**4+1)

[-1.63116471757557,
 0.0,
 0.0885580841208748,
 -1.0748505294265 - 0.665008885328002*I,
 -1.0748505294265 + 0.665008885328002*I]

There are 3 real roots but interest rate is generally positive. -1.63116471757557 and 0.0 are not realistic numbers for interest rate. Therefore there's only one IRR 8.85580841208748%.

Corporate Finance Exercise set 4 Question 7-13: 
to get all the possible IRRs through solving the polynomial equation
(5*x-0.9)*(1+x)^10+1 = 0, we run the following 3 lines of codes.

In [2]:
from sympy import Symbol, solve
x = Symbol("x")
solve((5*x-0.9)*(1+x)**10+1)

[-1.79531533426579,
 0.0274578401665965,
 0.108791830692669,
 -1.65735878173071 - 0.453700564327409*I,
 -1.65735878173071 + 0.453700564327409*I,
 -1.28662762317853 - 0.757299515247718*I,
 -1.28662762317853 + 0.757299515247718*I,
 -0.798175626017105 - 0.806508412941909*I,
 -0.798175626017105 + 0.806508412941909*I,
 -0.338305137370402 - 0.570661250186463*I,
 -0.338305137370402 + 0.570661250186463*I]

There are 3 real roots -1.79531533426579,  0.0274578401665965, 0.108791830692669. But cost of capital is positive. Therefore, there are only two IRRs, 2.745784% and 10.879183%. 